### PR TITLE
Just for viewing changes

### PR DESCRIPTION
**Pull request by nimanasiri**

### CODE REVIEW

File: snappy/snappy_compress.c [hunk 1]
```diff
@@ -484,14 +524,20 @@ snappy_status snappy_compress_host(struct host_buffer_context *input, struct hos
 	return SNAPPY_OK;
 }
 
-snappy_status snappy_compress_dpu(struct host_buffer_context *input, struct host_buffer_context *output, uint32_t block_size, struct program_runtime *runtime)
+snappy_status snappy_compress_dpu(const unsigned char *in, size_t in_len, unsigned char *out, size_t *out_len,
+            void *wrkmem, struct program_runtime *runtime)
 {
+	// Set block size
+	uint32_t block_size = 4 * 1024;
```
Comment (jnider) [inline snappy/snappy_compress.c]: This should be a global variable, or better, a define (the value should not change during the program)

File: test/alice.txt [hunk 2]
```diff
@@ -1,2 +1,14 @@
 Alice was beginning to get very tired of sitting by her sister on the bank, and of having nothing to do: once or twice she had peeped into the book her sister was reading, but it had no pictures or conversations in it, “and what is the use of a book,” thought Alice “without pictures or conversations?”
+Alice was beginning to get very tired of sitting by her sister on the bank, and of having nothing to do: once or twice she had peeped into the book her sister was reading, but it had no pictures or conversations in it, “and what is the use of a book,” thought Alice “without pictures or conversations?”
```
Comment (jnider) [inline test/alice.txt]: We have files of many lengths - please don't change Alice! You can use 'coding.txt' if you need something longer.

File: snappy/dpu_snappy.c [hunk 3]
```diff
@@ -210,8 +211,8 @@ int main(int argc, char **argv)
 			write_output_host(output_file, &output);
 
 		if (compress) {
-			printf("Compressed %ld bytes to: %s\n", output.length, output_file);
-			printf("Compression ratio: %f\n", 1 - (double)output.length / (double)input.length);
+			// printf("Compressed %ld bytes to: %s\n", output.length, output_file);
```
Comment (jnider) [inline snappy/dpu_snappy.c]: If you don't want these to be printed every time, add them to an #ifdef DEBUG section

File: snappy/Makefile [hunk 4]
```diff
@@ -1,5 +1,5 @@
 CC = gcc
-CFLAGS = --std=c99 -O3 -g -Wall -Wextra -I../PIM-common/common/include
+CFLAGS = --std=c99 -O0 -g -Wall -Wextra -I../PIM-common/common/include
```
Comment (jnider) [inline snappy/Makefile]: What's this about?

File: snappy/Makefile [hunk 5]
```diff
@@ -1,5 +1,5 @@
 CC = gcc
-CFLAGS = --std=c99 -O0 -g -Wall -Wextra -I../PIM-common/common/include
+CFLAGS = --std=c99 -O3 -g -Wall -Wextra -I../PIM-common/common/include
```
Comment (jnider) [inline snappy/Makefile]: ah, I see the two patches are really one - can you please merge them?

File: test/alice.txt (hunk 2 above)
Comment (nimanasiri) [reply test/alice.txt]: I just wanted something that's just under 4k, but will change this back.

**Pull request closed without merging**